setup.cfg

```diff
@@ -0,0 +1,42 @@
+[bumpversion]
+current_version = 0.0.0
+commit = True
+tag = True
+
+[bumpversion:file:docs/conf.py]
+search = release = "{current_version}"
+replace = release = "{new_version}"
+
+[bumpversion:file:setup.py]
+search = version="{current_version}"
+replace = version="{new_version}"
+
+[bumpversion:file:craft_providers/__init__.py]
+search = __version__ = "{current_version}"
+replace = __version__ = "{new_version}"
+
+[bdist_wheel]
+universal = 1
+
+[codespell]
+quiet-level = 3
+skip = .mypy_cache,.direnv,.git
+
+[flake8]
+exclude = docs venv .venv .mypy_cache .direnv .git __pycache__
+max-line-length = 88
+extend-ignore = E203, E501, W503
+
+[mypy]
+python_version = 3.8
+
+[pycodestyle]
+ignore = E402, E501, W503
+
+[pydocstyle]
+ignore = D105, D107, D203, D204, D213, D215
```

Comment:
Maybe add a comment on what each these exceptions are (for all ignores)

Reply:
I will do that.  FWIW most of these are already ignored in the default conventions:
http://www.pydocstyle.org/en/stable/error_codes.html#default-conventions

Reply:
added explanations, also removed/regenerated configs for other linters (will add explanations & disable as required)